index.bs

```diff
@@ -1097,6 +1098,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 :: A [=credential property=] is some characteristic property of a [=public key credential source=], such as whether it is a
     [=client-side discoverable credential=] or a [=server-side credential=].
 
+: <dfn>Hardware-bound Device Key Pair</dfn>
+: <dfn>Device-bound Key</dfn>
+: <dfn>Device Private Key</dfn>
+: <dfn>Device Public Key</dfn>
+:: A [=hardware-bound device key pair=], also known as a [=device-bound key=], is an [=authenticator=]-, [=[RP]=]-, and [=user credential=]-specific public key pair created upon a [=[RP]=]'s request via the [=devicePubKey=] [=WebAuthn extension=].
+    The [=authenticator=] that a [=hardware-bound device key pair=] is created upon guarantees that the [=device private key=] is securely stored in hardware, i.e., it is unextractable. See also [[#sctn-device-publickey-extension]].
+
```

Comment:
Since this section states "The authenticator [...] guarantees that the device private key is securely stored in hardware", should we emphasize that this guarantee is only realized with a trusted attestation? So we don't lead RPs to believe that they get this guarantee from just using the extension.

Maybe something like:

```suggestion
: <dfn>Hardware-bound Device Key Pair</dfn>
: <dfn>Device-bound Key</dfn>
: <dfn>Device Private Key</dfn>
: <dfn>Device Public Key</dfn>
:: A [=hardware-bound device key pair=], also known as a [=device-bound key=], is an [=authenticator=]-, [=[RP]=]-, and [=user credential=]-specific public key pair created upon a [=[RP]=]'s request via the [=devicePubKey=] [=WebAuthn extension=].
    The [=authenticator=] that a [=hardware-bound device key pair=] is created upon guarantees that the [=device private key=] is securely stored in hardware, i.e., it is unextractable. See also [[#sctn-device-publickey-extension]].

    Note: All guarantees about [=authenticators'=] operation rely on [=attestation=]. In particular, [=[RPS]=] MUST NOT rely on the above guarantee of unextractability unless supported by a valid, trusted [=attestation statement=].
```

Reply:
We may also need to clarify which attestation (I suspect it is the dpk attestation).   DPK may be created on different hardware platforms so one credential might have a DPK on one platform that is hardware backed but the same credential on a different device may have a DPK that is only in software.

Reply:
> [a user] credential might have a DPK on one platform that is hardware backed but the same [user] credential on a different device may have a **DPK that is only in software**.  (emphasis added)

That is an interesting point. It raises the question of whether we ought to be using terms such as "hardware-bound", "stored in hardware", "only in software", because these are all implementation characteristics and their delineation can be fuzzy.  The above definition seems naively too strident.

Perhaps a more evenhanded characterization of DPK intention is that the device private key is "device-bound" (in some fashion) and that its extraction is intended to be difficult or implausible?

This would imply changing the [=hardware-bound device key pair=] term to, say, [=device-bound key pair=], not using the term "guarantees"  (I've been thinking that term is too strong, in any case), and softening the notion of "unextractable".  Perhaps like so:
```
: <dfn>Device-bound Key Pair</dfn>
: <dfn>Device-bound Key</dfn>
: <dfn>Device Private Key</dfn>
: <dfn>Device Public Key</dfn>

:: A [=device-bound key pair=], also known as a [=device-bound key=], is an 
    [=authenticator=]-, [=[RP]=]-, and [=user credential=]-specific public key pair 
    created upon [=[RP]=] request via the [=devicePubKey=] [=WebAuthn extension=]. 

    [=Device-bound key pairs=] are intended to be implemented by 
    [=authenticators=] such that the [=device private key=] is securely bound to 
    the [=authenticator=], i.e., extraction of the [=device private key=] is difficult or 
    implausible. See also [[#sctn-device-publickey-extension]].
```
I'd re-characterize the suggested Note re "guarantees" to be something like this:
    Note: Assurances about [=authenticators'=] operation rely on [=attestation=]. Along with [[#sctn-device-publickey-extension-verification|verifying]] [=device public key attestation object|device public keys' attestation objects=], [=[RPS]=] can input associated metadata into their risk-analysis calculations.  For example, the FIDO Metadata Service [[FIDOMetadataService]] provides one way to access such information.

...and add the note down in the [[#sctn-device-publickey-extension-usage]]  section (which is in the DPK extension definition section).

Reply:
As suggested by @emlun , I think the "guarantee" statement needs qualification with attestation.